CRM/Contribute/Form/UpdateBilling.php

```diff
@@ -47,14 +47,13 @@ public function preProcess() {
     if ($this->_mid) {
       $this->_paymentProcessor = CRM_Financial_BAO_PaymentProcessor::getProcessorForEntity($this->_mid, 'membership', 'info');
       $this->_paymentProcessor['object'] = CRM_Financial_BAO_PaymentProcessor::getProcessorForEntity($this->_mid, 'membership', 'obj');
-      $this->_subscriptionDetails = CRM_Contribute_BAO_ContributionRecur::getSubscriptionDetails($this->_mid, 'membership');
```

Comment:
@mattwire are you sure this isn't altering what is stored in `$this->_subscriptionDetails ` - my brain failed before making sense of that....

Reply:
Yes. parent::preProcess()` gets/sets recur ID, contribution ID and membership ID if available. Then calls `CRM_Contribute_BAO_ContributionRecur::getSubscriptionDetails()` which retrieves the recur information.

- If we pass in only the membership ID there is no way it could change.
- If we pass in the recur ID there is no way it could change.
- If we pass in only contribution ID we don't hit the `if ($this->_mid`)` so it doesn't change.


If we pass in contribution ID and membership ID then we're using the `civicrm_membership_payment` table and there's a possibility of returing a different recur. But that's a completely unsupported configuration because we'd have to have two recurring contributions making payments on the same membership and that has never been supported in CiviCRM (we do partially support the opposite which is two memberships on the same recurring contribution but that scenario is not changed here).

Reply:
OK - my brain won't process that but I feel confident you have worked through it